code/game/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,182 @@
+#define ELECTROLYZER_MODE_STANDBY	"standby"
+#define ELECTROLYZER_MODE_WORKING	"working"
+
+/obj/machinery/electrolyzer
+	anchored = FALSE
+	density = TRUE
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_WIRES_IF_OPEN
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "electrolyzer-off"
+	name = "space electrolyzer"
+	desc = "Thanks to the fast and dynamic response of our electrolyzers, on-site hydrogen production is guaranteed. Warranty void if used by clowns"
+	max_integrity = 250
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 80, "acid" = 10)
+	circuit = /obj/item/circuitboard/machine/electrolyzer
+	ui_x = 400
+	ui_y = 305
+
+	var/obj/item/stock_parts/cell/cell
+	var/on = FALSE
+	var/mode = ELECTROLYZER_MODE_STANDBY
+	var/workingPower = 1
+	var/efficiency = 0.5
+
+/obj/machinery/electrolyzer/get_cell()
+	return cell
+
+/obj/machinery/electrolyzer/Initialize()
+	. = ..()
+	cell = new(src)
+	update_icon()
+
+/obj/machinery/electrolyzer/on_construction()
+	qdel(cell)
+	cell = null
+	panel_open = TRUE
+	update_icon()
+	return ..()
+
+/obj/machinery/electrolyzer/on_deconstruction()
+	if(cell)
+		component_parts += cell
+		cell = null
+	return ..()
+
+/obj/machinery/electrolyzer/examine(mob/user)
+	. = ..()
+	. += "\The [src] is [on ? "on" : "off"], and the hatch is [panel_open ? "open" : "closed"]."
+	if(cell)
+		. += "The charge meter reads [cell ? round(cell.percent(), 1) : 0]%."
+	else
+		. += "There is no power cell installed."
+
+/obj/machinery/electrolyzer/update_icon_state()
+	if(on)
+		icon_state = "electrolyzer-[mode]"
+	else
+		icon_state = "electrolyzer-off"
+
+/obj/machinery/electrolyzer/update_overlays()
+	. = ..()
+
+	if(panel_open)
+		. += "electrolyzer-open"
+
+/obj/machinery/electrolyzer/process()
+	if(!on || !is_operational())
+		if (on) // If it's broken, turn it off too
+			on = FALSE
+		return PROCESS_KILL
+
+	if(cell && cell.charge > 0)
+		var/turf/L = loc
+		if(!istype(L))
+			if(mode != ELECTROLYZER_MODE_STANDBY)
+				mode = ELECTROLYZER_MODE_STANDBY
+				update_icon()
+			return
+
+		var/datum/gas_mixture/env = L.return_air() //get air from the turf
+		var/datum/gas_mixture/removed
+		removed = env.remove(0.1 * env.total_moles())
```

Comment:
This should just be `var/datum/gas_mixture/removed = env.remove(0.1 * env.total_moles())`.

Reply:
done